src/ServiceControl/HeartbeatMonitoring/HeartbeatsMonitor.cs

```diff
@@ -147,5 +149,114 @@ void Initialise()
                 }
             }
         }
+
+        class HeartbeatsWriter : FeatureStartupTask
+        {
+            private readonly IDocumentStore store;
+            public IBus Bus { get; set; }
+
+            public HeartbeatStatusProvider HeartbeatStatusProvider { get; set; }
+
+            public HeartbeatsWriter(IDocumentStore store)
+            {
+                this.store = store;
+            }
+
+            protected override void OnStart()
+            {
+                timer = new Timer(Refresh, null, TimeSpan.FromSeconds(15), Timeout.InfiniteTimeSpan);
+            }
+
+            protected override void OnStop()
+            {
+                using (var manualResetEvent = new ManualResetEvent(false))
+                {
+                    timer.Dispose(manualResetEvent);
+
+                    manualResetEvent.WaitOne();
+                }
+            }
+
+            void Refresh(object _)
+            {
+                Sync();
+
+                try
+                {
+                    timer.Change(TimeSpan.FromMinutes(2), Timeout.InfiniteTimeSpan);
+                }
+                catch (ObjectDisposedException)
+                {
+                }
+            }
+
+            void Sync()
+            {
+                using (var session = store.OpenSession())
+                {
+                    var storedHeartbeats = session.Query<Heartbeat>().ToDictionary(h=> h.Id);
+                    var inmemoryHeartbeats = HeartbeatStatusProvider.HeartbeatsPerInstance;
+
+                    foreach (var memHeartbeat in inmemoryHeartbeats)
+                    {
+                        Heartbeat heartbeat;
+                        if (storedHeartbeats.TryGetValue(memHeartbeat.Key, out heartbeat))
+                        {
+                            if (heartbeat.Disabled)
+                            {
+                                continue;
+                            }
```

Comment:
This means that we don't update the details of heartbeats when we aren't monitoring them. Were we before? Should we?